Add upload and download functions to user guide (#34)
exasol/bucketfs-utils-python: doc/user_guide/upload_download_functions.rst

**********************************************
Uploading and downloading to/from the BucketFS
**********************************************

With these functions you can upload and download data to and from the BucketFS.
The upload and download functions can have different sources or targets,
such as files, file-objects, strings or Python objects.
For more details, please refer to our

 * :doc:`Upload API </api/exasol_bucketfs_utils_python.upload>`
 * :doc:`Download API </api/exasol_bucketfs_utils_python.download>`

################################################
Uploading and downloading data from and to files
################################################

> The simplest form of the download and upload functions simply uploads a local file to the BucketFS or
> download the data from the BucketFS into a local file.

Reviewer: ```suggestion
In their simplest form the download and upload functions simply upload a local file to the BucketFS or
download the data from the BucketFS into a local file.
```
